Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Undo/PrefabUndo.cpp

```diff
@@ -121,7 +121,7 @@ namespace AzToolsFramework
                 {
                     // Create a copy of the DOM of the end state so that it shares the lifecycle of the cached DOM.
                     PrefabDom endStateCopy;
-                    endStateCopy.CopyFrom(endState, cachedDom->get().GetAllocator());
+                    endStateCopy.CopyFrom(endState, endStateCopy.GetAllocator());
```

Comment:
(note from author) I'm actually curious as to why this was that way in the first place, the comment is a bit confusing.
In general, the allocator you want to use is the allocator of the DOM itself, not someone else...

Reply:
I'm not sure what your code above refers to for cachedOwningInstanceDom, or entityDom, neither of those are present in the Capture function as of yet?

Reply:
Do you perhaps mean this instead?
```
                    PrefabDomValue endStateCopy(endState, cachedDom->get().GetAllocator());
                    // Update the cached instance DOM corresponding to the entity so that the same modified entity isn't reloaded again.
                    PrefabDomPath entityPathInDom(entityAliasPath.c_str());
                    entityPathInDom.Set(cachedDom->get(), endStateCopy.Move());
```

Reply:
oh sorry I was referring to some code elsewhere. Yeah, this is what I mean.

Reply:
I think thats worse, perhaps I don't understand. Looking at the code, here's my analysis
```
// the following line makes a deep copy of endState into endStateCopy and allocates memory to do so from the allocator
// from cachedDom.  cachedDom's allocator thus would grow by the size of endState's values.
PrefabDomValue endStateCopy(endState, cachedDom->get().GetAllocator());

PrefabDomPath entityPathInDom(entityAliasPath.c_str());

// important note - rapidjson Pointer.Set(x, y) does not have a move form that takes &&y, only &y, and const &y
// so this invokes GenericValue.Set(non-const GenericDocument&, non-const Value&), not value&& which
// removes the rvalue-reference from value.
// which itself invokes Pointer::Create(document) = value.
// The call to Create ensures that all parent values exist, so it uses cachedDom's allocator to create them if necessary.
// cachedDom's memory will grow by that amount.
// the assignment operator is then called on the new child, without a rvalue reference.
// this = operator then just deep copies the value
 entityPathInDom.Set(cachedDom->get(), endStateCopy.Move());

 // the end result is that cachedDom's allocator grows by the size of endstate,
 // plus the size of the elements required to create the tree of parents to match the pointer
// plus the size ofendState again.
```

Reply:
Ok, so it turns out GenericValue::operator=(GenericValue& other) adopts the pointers of other (without destroying other).
This means the only option here that mkes sense is to copy into the final dom.
This can be achieved by the one liner
```
 entityPathInDom.Set(cachedDom->get(), endState);
 ```
because endState is already a const& and thus will already trigger a copy into the dom using the dom's allocator

Reply:
discussed offline. We can avoid doing the copy by ourselves and let Set() do it. But in this case we need to be careful about the type of endState. It needs to be const& (not non-const &) to trigger the copy version of Set().